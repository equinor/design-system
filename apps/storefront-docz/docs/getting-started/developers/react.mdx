---
title: Developers
tabs:
  - Introduction
  - CSS framework
  - React
  - Figma
mode: draft
route: /getting-started/developers/react/
---

:warning: Information coming soon
