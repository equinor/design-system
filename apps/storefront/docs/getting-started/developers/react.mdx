---
title: Developers
metaTitle: Developers react
tabs:
  - Introduction
  - CSS framework
  - React
  - Figma
mode: draft
route: /getting-started/developers/react/
---

:warning: Information coming soon
