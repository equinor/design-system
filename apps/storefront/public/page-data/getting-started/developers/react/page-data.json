{"componentChunkName":"component---docs-getting-started-developers-react-mdx","path":"/getting-started/developers/react/","result":{"pageContext":{"frontmatter":{"title":"Developers","metaTitle":"Developers react","tabs":["Introduction","CSS framework","React","Figma"],"mode":"draft","route":"/getting-started/developers/react/"},"entry":{"id":"6f9784da00517d57e4627269567073e6","filepath":"docs/getting-started/developers/react.mdx","fullpath":"/Users/vnys/Kode/storefront-replace/apps/storefront/docs/getting-started/developers/react.mdx","route":"/getting-started/developers/react/","slug":"docs-getting-started-developers-react","name":"React","menu":"","headings":[]}}}}