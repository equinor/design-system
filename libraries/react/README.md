@eds-ui/react
===============

1. `npm i -g pnpm`
1. `pnpm install`
1. `pnpm recursive install`

## Run storybook

`pnpm start`

## Build modules

`pnpm recursive run build`
