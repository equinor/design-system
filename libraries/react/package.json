{
  "name": "@eds-ui/react",
  "private": true,
  "version": "0.0.1",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "jest",
    "storybook": "start-storybook"
  },
  "keywords": [],
  "author": "",
  "license": "AGPL",
  "dependencies": {
    "react": "^16.8.6",
    "react-dom": "^16.8.6"
  },
  "devDependencies": {
    "@babel/cli": "^7.4.4",
    "@babel/core": "^7.4.4",
    "@babel/preset-env": "^7.4.4",
    "@babel/preset-react": "^7.0.0",
    "@storybook/react": "^5.0.11",
    "babel-plugin-styled-components": "^1.10.0",
    "enzyme": "^3.9.0",
    "enzyme-adapter-react-16": "^1.12.1",
    "jest": "^24.8.0",
    "jest-styled-components": "^6.3.1",
    "rollup": "^1.11.3",
    "rollup-plugin-babel": "^4.3.2",
    "rollup-plugin-commonjs": "^9.3.4",
    "rollup-plugin-node-resolve": "^4.2.3",
    "styled-components": "^4.2.0"
  },
  "peerDependencies": {
    "react": "^16.8.6",
    "react-dom": "^16.8.6",
    "styled-components": "^4.2.0",
    "prop-types": "^15.7.2"
  },
  "jest": {
    "verbose": true,
    "setupFilesAfterEnv": [
      "<rootDir>/testSetup.js"
    ],
    "testPathIgnorePatterns": [
      "<rootDir>/(node_modules)"
    ],
    "transformIgnorePatterns": [
      "<rootDir>/(node_modules)"
    ],
    "haste": {
      "providesModuleNodeModules": [
        ".*"
      ]
    },
    "roots": [
      "packages/node_modules"
    ]
  }
}
