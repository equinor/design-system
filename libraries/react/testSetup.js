import React from 'react'
import Enzyme from 'enzyme'
import 'jest-styled-components'
import styled from 'styled-components'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })
