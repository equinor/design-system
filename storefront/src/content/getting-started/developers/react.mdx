---
title: React
tabs:
    - Introduction
    - CSS framework
    - React
    - Figma
    - Contributing
mode: draft
---


:warning: Information coming soon
