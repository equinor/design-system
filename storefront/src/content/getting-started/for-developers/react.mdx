---
title: React
tabs:
    - Introduction
    - CSS framework
    - React
    - Figma
    - Contributing
---

Developers React
