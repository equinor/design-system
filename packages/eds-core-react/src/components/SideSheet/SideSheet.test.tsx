import { render, cleanup, screen, act } from '@testing-library/react'
import '@testing-library/jest-dom'
import 'jest-styled-components'
import { axe } from 'jest-axe'
import styled from 'styled-components'
import { SideSheet } from '.'

const StyledSidesheet = styled(SideSheet)`
  position: relative;
  height: 100px;
  width: 100px;
`

afterEach(cleanup)

describe('SideSheet', () => {
  it('Matches snapshot', () => {
    const { asFragment } = render(<SideSheet variant="large" title="Title" />)
    expect(asFragment()).toMatchSnapshot()
  })
  it('Should pass a11y test', async () => {
    const { container } = render(<SideSheet title="Title" />)
    await act(async () => {
      const results = await axe(container)
      expect(results).toHaveNoViolations()
    })
  })
  it('Has correct width', () => {
    render(<SideSheet variant="large" title="Title" data-testid="sidesheet" />)
    expect(screen.getByTestId('sidesheet')).toHaveStyleRule('width', '480px')
  })
  it('Has provided necessary props', () => {
    const title = 'Title'
    const variant = 'large'
    render(<SideSheet variant={variant} title={title} />)
    expect(screen.getByText(title)).toBeDefined()
  })
  it('Can extend the css for the component', () => {
    render(<StyledSidesheet data-testid="sidesheet" />)
    const sidesheet = screen.getByTestId('sidesheet')
    expect(sidesheet).toHaveStyleRule('position', 'relative')
    expect(sidesheet).toHaveStyleRule('height', '100px')
    expect(sidesheet).toHaveStyleRule('width', '100px')
  })
})
