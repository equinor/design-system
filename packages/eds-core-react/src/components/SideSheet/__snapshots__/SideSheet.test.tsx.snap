// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideSheet Matches snapshot 1`] = `
<DocumentFragment>
  .c2 {
  margin: 0;
  color: var(--eds_heading__h2_color,rgba(61,61,61,1));
  font-family: Equinor;
  font-size: 1.750rem;
  font-weight: 400;
  line-height: 1.250em;
  text-align: left;
}

.c4 {
  display: grid;
  grid-gap: var(--eds_button__gap,8px);
  grid-auto-flow: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 > :is(svg,img) {
  margin-top: var(--eds_button__icon__margin_y,0);
  margin-bottom: var(--eds_button__icon__margin_y,0);
}

.c3 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  position: relative;
  cursor: pointer;
  display: inline-block;
  background: transparent;
  height: 40px;
  width: 40px;
  padding-left: 0;
  padding-top: var(--eds_button__padding_y,0);
  padding-right: 0;
  padding-bottom: var(--eds_button__padding_y,0);
  border: 0px solid transparent;
  border-radius: 50%;
  margin: 0;
  color: var(--eds_interactive_primary__resting,rgba(0,112,121,1));
  font-family: Equinor;
  font-size: var(--eds_button__font_size,0.875rem);
  font-weight: 500;
  line-height: 1.143em;
  text-align: center;
}

.c3 svg {
  justify-self: center;
}

.c3::before {
  position: absolute;
  top: 0;
  left: 0;
  width: auto;
  min-height: auto;
  content: '';
}

.c3::after {
  position: absolute;
  top: -0;
  left: -4px;
  width: 48px;
  height: 48px;
  content: '';
}

.c3:focus {
  outline: none;
}

.c3[data-focus-visible-added]:focus {
  outline: 2px dashed rgba(0,112,121,1);
  outline-offset: -2px;
}

.c3:focus-visible {
  outline: 2px dashed rgba(0,112,121,1);
  outline-offset: -2px;
}

.c3::-moz-focus-inner {
  border: 0;
}

.c3:disabled,
.c3[aria-disabled='true'] {
  cursor: not-allowed;
  background: transparent;
  border: 1px solid transparent;
  margin: 0;
  color: var(--eds_interactive__disabled__text,rgba(190,190,190,1));
  font-family: Equinor;
  font-size: 0.875rem;
  font-weight: 500;
  line-height: 1.143em;
  text-align: center;
}

.c0 {
  height: 100%;
  position: absolute;
  z-index: 1200;
  top: 0;
  right: 0;
  box-sizing: border-box;
  background: var(--eds_ui_background__default,rgba(255,255,255,1));
  width: 480px;
  border-left: 2px solid var(--eds_ui_background__light,rgba(247,247,247,1));
  padding-left: 14px;
  padding-top: 16px;
  padding-right: 16px;
  margin: 0;
  color: var(--eds_paragraph__body_short_color,rgba(61,61,61,1));
  font-family: Equinor;
  font-size: 1.000rem;
  font-weight: 400;
  line-height: 1.250em;
  text-align: left;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-bottom: 24px;
  padding-right: 10px;
}

@media (hover:hover) and (pointer:fine) {
  .c3:hover {
    background: var(--eds_interactive_primary__hover_alt,rgba(222,237,238,1));
    border: 0px solid transparent;
    border-radius: 50%;
  }
}

@media (hover:hover) and (pointer:fine) {
  .c3:disabled:hover,
  .c3[aria-disabled='true']:hover {
    background: transparent;
  }
}

<div
    class="c0"
    width="480px"
  >
    <div
      class="c1"
    >
      <h2
        class="c2"
      >
        Title
      </h2>
      <button
        aria-label="Close sidesheet"
        class="c3"
        tabindex="0"
        title="Close"
        type="button"
      >
        <span
          class="c4"
        >
          <svg
            aria-hidden="true"
            class=""
            fill="currentColor"
            height="24px"
            viewBox="0 0 24 24"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class=""
              clip-rule="evenodd"
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"
              data-testid="eds-icon-path"
              fill-rule="evenodd"
              height="24"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
